### DISCUSSION TITLE
Is there a way to view current memory usages inside of JavaScript?

### DISCUSSION THREAD

**Discussion by damionbroadaway** [mozilla/rhino#1272] [Q&A]
_I realize this may not be the best place for this but I am having trouble finding any resources on the topic. Everything I can find assumes you are developing with Rhino._

We use a product that uses Rhino to run the .js scripts we write. So, all we can do is write the JS that gets uploaded into their system.

Do we have access to anything in JS that can help us troubleshoot? Specifically in regards to memory?

The provider has told us one script is running out of memory and that's it. In Node we have access to `process`. Is there anything in Rhino that we can call that's similar? Any type of debug object that we could dump or invoke or anything? Even if it is dependent on the provider enabling a feature or whatever.

Anything will be appreciated.

**Comment by zhufeizzz** [comment]
Same question, can anyone give some advice?
